src/test/java/com/segment/analytics/android/integration/firebase/FirebaseTest.java

```diff
@@ -1,5 +1,148 @@
+
+    @Before
+    public void setUp() {
+        firebase = PowerMockito.mock(FirebaseAnalytics.class);
+        Context context = PowerMockito.mock(Context.class);
+
+        PowerMockito.mockStatic(FirebaseAnalytics.class);
+        Mockito.when(FirebaseAnalytics.getInstance(context)).thenReturn(firebase);
+
+        integration = new FirebaseIntegration(context, Logger.with(VERBOSE));
+    }
+
+    @Test
+    public void identify() {
+        Traits traits = createTraits("foo");
+        integration.identify(new IdentifyPayloadBuilder().traits(traits).build());
+        verify(firebase).setUserId("foo");
+    }
+
+    @Test
+    public void identifyWithTraits() {
+        Traits traits = createTraits("foo")
+                .putAge(20)
+                .putFirstName("bar")
+                .putLastName("baz")
+                .putValue("anonymousId", 123)
+                .putValue("Sign Up Date", new Date())
+                .putValue("  extra spaces        ", "bar");
+
+
+        integration.identify(new IdentifyPayloadBuilder().traits(traits).build());
+
+        verify(firebase).setUserId("foo");
+        verify(firebase).setUserProperty("firstName", "bar");
+        verify(firebase).setUserProperty("lastName", "baz");
+        verify(firebase).setUserProperty("anonymousId", "123");
+        verify(firebase).setUserProperty("Sign_Up_Date", "2017-09-07");
+        verify(firebase).setUserProperty("extra_spaces", "bar");
+    }
+
+    @Test
+    public void track() {
+        integration.track(new TrackPayloadBuilder().event("foo").build());
+        verify(firebase).logEvent(eq("foo"), bundleEq(new Bundle()));
+    }
+
+    @Test
+    public void trackWithProperties() {
+        Properties properties = new Properties()
+                .putValue("integer", 1)
+                .putValue("double", 1.0)
+                .putValue("string", "foo")
+                .putValue("date", new Date(117, 0, 1))
+                .putValue("key with spaces", "bar")
+                .putValue("total", 100.0)
+                .putValue("  extra spaces   ", "baz");
+
+        integration.track(new TrackPayloadBuilder().properties(properties).event("foo").build());
+
+        Bundle expected = new Bundle();
+        expected.putInt("integer", 1);
+        expected.putDouble("double", 1.0);
+        expected.putString("string", "foo");
+        expected.putString("date", "2017-01-01");
+        expected.putString("key_with_spaces", "bar");
+        expected.putDouble("value", 100.0);
+        expected.putString("currency", "USD");
+        expected.putString("extra_spaces", "baz");
+
+        verify(firebase).logEvent(eq("foo"), bundleEq(expected));
+    }
+
+    @Test
+    public void testMakeKey() {
+        String string = " long event name with random spaces that is more than forty characters long           ";
+        assertThat(integration.makeKey(string)).isEqualTo("long_event_name_with_random_spaces_that_");
```

Comment:
This is a static function. It should be used as `FirebaseIntegration.makeKey...` rather than as an instance.